pkg/watch/mux.go

```diff
@@ -149,6 +151,7 @@ func (m *Broadcaster) Action(action EventType, obj runtime.Object) {
 // You must not call Action after calling Shutdown.
```

Comment:
Outbound channels could be buffered, and even if they weren't, you don't wait for close to have been called on them.